x-pack/plugin/search-business-rules/src/main/java/org/elasticsearch/xpack/searchbusinessrules/PinnedQueryBuilder.java

```diff
@@ -129,7 +126,7 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         );
 
         static {
-            PARSER.declareString(constructorArg(), INDEX_FIELD);
+            PARSER.declareStringOrNull(optionalConstructorArg(), INDEX_FIELD);
```

Comment:
We also need to update this based on the version.

Reply:
`declareStringOrNull`

This means we allow the value to be string or explicitly `null`. I think we instead allow it to be `string` or not provided at all. Consequently, it should just be `PARSER.declareString(optionalConstructorArg(), INDEX_FIELD);`